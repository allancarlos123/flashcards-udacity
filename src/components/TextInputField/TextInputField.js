import React from 'react';
import styled from 'styled-components/native'
import { View } from 'react-native';

const Label = styled.Text ``;

const Input = styled.TextInput `
  height: 40;
`

const TextInputField = ({
  label,
  maxLength,
  placeholder,
  autoFocus,
  value,
  onChangeText,
}) => (
  <View>
    <Label>{label}</Label>
    <Input
      placeholder={placeholder}
      maxLength={maxLength}
      selectionColor = '#ED4E5F'
      autoCorrect={false}
      autoFocus={autoFocus}
      keyboardType="default"
      value={value}
      onChangeText={onChangeText}
      underlineColorAndroid="transparent"
      clearButtonMode="always"
    />
  </View>
)

export default TextInputField