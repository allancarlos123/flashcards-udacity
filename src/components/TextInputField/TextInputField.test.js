import React from 'react';
import { create } from 'react-test-renderer';

import TextInputField from './';

describe('TextInputField', () => {
  const onChangeText = jest.fn();
  const tree = create(
    <TextInputField
    placeholder="something"
    maxLength={30}
    label="Enter something"
    onChangeText={onChangeText}
    />
  );
  
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot()
  });
});