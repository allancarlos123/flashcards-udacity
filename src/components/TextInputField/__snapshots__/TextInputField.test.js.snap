// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputField renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    Enter something
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    autoFocus={undefined}
    clearButtonMode="always"
    keyboardType="default"
    maxLength={30}
    onChangeText={[MockFunction]}
    placeholder="something"
    selectionColor="#ED4E5F"
    style={
      Array [
        Object {
          "height": 40,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value={undefined}
  />
</View>
`;
